src/Microsoft.AspNetCore.SignalR.Core/Internal/DefaultHubProtocolResolver.cs

```diff
@@ -11,23 +14,33 @@ namespace Microsoft.AspNetCore.SignalR.Internal
     public class DefaultHubProtocolResolver : IHubProtocolResolver
     {
         private readonly IOptions<HubOptions> _options;
+        private readonly ILogger<DefaultHubProtocolResolver> _logger;
+        private readonly Dictionary<string, IHubProtocol> _availableProtocols;
 
-        public DefaultHubProtocolResolver(IOptions<HubOptions> options)
+        public DefaultHubProtocolResolver(IOptions<HubOptions> options, IEnumerable<IHubProtocol> availableProtocols, ILogger<DefaultHubProtocolResolver> logger)
         {
             _options = options;
+            _logger = logger;
+            _availableProtocols = availableProtocols.ToDictionary(p => p.Name, StringComparer.OrdinalIgnoreCase);
```

Comment:
This will fail kinda poorly if somebody the same name is used. We should probably manually overwrite.

Reply:
Sure, I had the same thought but was just being lazy :). I'll put in a simple check to throw a useful exception if there are multiple with the same name.